### PR TITLE
add support for Elixir in repairability analysis

### PR DESCRIPTION
**Pull request by niloofartrg**
Implementation of Elixir repair tool.

### CODE REVIEW

File: src/main/java/fr/inria/coming/repairability/RepairTools.java [hunk 1]
```diff
@@ -26,7 +26,8 @@
             "Nopol",
             "JKali",
             "NPEfix",
-            "JGenProg"
+            "JGenProg",
+            "elixir"
```
Comment (sedflix) [inline src/main/java/fr/inria/coming/repairability/RepairTools.java]: Capitalize the first letter to maintain the convention?

File: src/main/java/fr/inria/coming/repairability/repairtools/elixir.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+ * <p>
+ * 5.skip statement
+ * 6.return a null to caller
+ * 7.return a new object to caller
+ * 8.return an existing compatible object to caller
+ * 9.return to caller (void method)
+ * <p>
+ * I added two pattern files which checks for a variable being inserted or updated (1-4) and another one to check for insertion of return. (5-9)
+ */
+public class elixir extends AbstractRepairTool {
```
Comment (sedflix) [inline src/main/java/fr/inria/coming/repairability/repairtools/elixir.java]: elixir -> Elixir

File: src/main/java/fr/inria/coming/repairability/repairtools/elixir.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+    private static final String[] patternFileNames = {
+            "ep_1.xml",
+            "ep_2.xml",
+            "ep_3.xml",
+            "ep_4.xml"
+    };
+
+    /**
+     * Encodes the search space of JMutRepair
```
Comment (sedflix) [inline src/main/java/fr/inria/coming/repairability/repairtools/elixir.java]: `JMutRepair` in this doc is odd

File: src/test/java/fr/inria/coming/spoon/repairability/repairtools/elixir.java [hunk 4]
```diff
@@ -0,0 +1,14 @@
+package fr.inria.coming.spoon.repairability.repairtools;
+
+
+import fr.inria.coming.changeminer.entity.FinalResult;
+import fr.inria.coming.spoon.repairability.TestUtills;
+import org.junit.Test;
+
+public class elixir {
```
Comment (sedflix) [inline src/test/java/fr/inria/coming/spoon/repairability/repairtools/elixir.java]: elixir -> Elixir and the Test Class naming convention should be followed.

File: src/main/resources/repairability/elixir/ep_1.xml [hunk 5]
```diff
@@ -0,0 +1,6 @@
+<pattern name="any_statement_deep">
```
Comment (sedflix) [inline src/main/resources/repairability/elixir/ep_1.xml]: name of all xml are same, ie, `any_statement_deep`.
File and pattern_name renaming should look cool

Comment (niloofartrg) [reply src/main/resources/repairability/elixir/ep_1.xml]: Thank you for your comments Siddharth.

File: src/main/java/fr/inria/coming/repairability/repairtools/Elixir.java [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+import fr.inria.coming.changeminer.util.PatternXMLParser;
+import gumtree.spoon.diff.operations.Operation;
+import spoon.reflect.code.CtBinaryOperator;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * NPEfix fixes the program by :
```
Comment (monperrus) [inline src/main/java/fr/inria/coming/repairability/repairtools/Elixir.java]: NPEfix? update doc

File: src/main/resources/repairability_test_files/Elixir/elixir2/two/elixir2_two_s.java [hunk 7]
```diff
@@ -0,0 +1,13 @@
+
+class JKali {
```
Comment (monperrus) [inline src/main/resources/repairability_test_files/Elixir/elixir2/two/elixir2_two_s.java]: should be called "elixir_something"

**Comment by monperrus** [comment]
Thanks!

**Pull request merged**